polaris-core/src/main/java/org/apache/polaris/core/storage/StorageLocation.java

```diff
@@ -41,16 +44,21 @@ public static StorageLocation of(String location) {
   protected StorageLocation(@Nonnull String location) {
     if (location == null) {
       this.location = null;
-    } else if (location.startsWith("file:/") || location.startsWith("/")) {
-      if (!location.startsWith(LOCAL_PATH_PREFIX)) {
-        String[] parts = location.split("/", -1);
-        // Converting "file:////abcd" through Hadoop Path and Java URI will still yield "file:///abcd"
-        parts[0] = LOCAL_PATH_PREFIX;
-        location = String.join("/", parts);
-      }
-      this.location = new Path(location).toUri().toString();
+      return;
+    }
+    String scheme = scheme(location);
+    String path;
+    if (scheme == null) {
+      path = location;
+      scheme = "file";
     } else {
-      this.location = new Path(location).toString();
+      path = location.substring(scheme.length() + 1);
+    }
+    path = normalizePath(path);
+    if (scheme.equals("file")) {
```

Comment:
ditto, let's at least put this in a const

Reply:
Roger that, that's reasonable.